nixos/modules/services/web-servers/keter/default.nix

```diff
@@ -0,0 +1,163 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.services.keter;
+in
+{
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ jappie ];
+  };
+  options.services.keter = {
+    enable = lib.mkEnableOption "Keter";
+    keterRoot = lib.mkOption {
+      type = lib.types.str;
+      default = "/var/lib/keter";
+      description = ''Mutable state folder for keter. In here the bundles get extracted,
+                      logs get written and the app's work folder is set to'';
+    };
+    keterPackage = lib.mkOption {
+      type = lib.types.package;
+      default = null;
+      description = ''set the keter package to be used, usually pkgs.haskellPackages.keter'';
+    };
+
+    globalKeterConfig = lib.mkOption {
+      type = lib.types.attrs;
+      default = {
+        ip-from-header = true;
+        listeners = [{
+          host = "*4";
+          port = 81;
+        }];
+      };
+      description = ''root is required to function in nixos .
+                      port 81 is set to default to allow nginx to reverse proxy into keter,
+                      so that nginx can do ssl management.
+                      You want that ip-from-header in the nginx setup case
+                      so it's not set to 127.0.0.1.
+                      Keep in mind root is overwritten by keterRoot.
+                      which is required for the nix derivation.
+                      '';
```

Comment:
Just say something like "Global configuration for keter", maybe add a `longDescription` and definitely add a `defaultText = lib.literalExpression "[{ port = 81; TODO fill this in }]"`

Reply:
I did this